controllers/component_image_controller.go

```diff
@@ -44,7 +44,7 @@ const (
 	ImageAnnotationName         = "image.redhat.com/image"
 	GenerateImageAnnotationName = "image.redhat.com/generate"
 
-	ImageRepositoryFinalizer = "image-controller.appstudio.openshift.io/image-repository"
+	ImageRepositoryComponentFinalizer = "image-controller.appstudio.openshift.io/image-repository"
```

Comment:
Are you planning to use this CRD for components as well? It would remove a lot of duplicate code.